lsp-rust.el

```diff
@@ -425,6 +430,57 @@ PARAMS progress report notification data."
                          (* (lsp--client-priority (gethash server lsp-clients)) -1)))
       (message (format "Switched to server %s." server)))))
 
+;; inlay hints
+
+(defvar-local lsp-rust-analyzer-inlay-hints-timer nil)
+
+(defun lsp-rust-analyzer-update-inlay-hints (buffer)
+  (if (and (lsp-rust-analyzer-initialized?)
+           (eq buffer (current-buffer)))
+      (lsp-send-request-async
```

Comment:
can you switch to lsp-request-async? lsp-send-request-async is deprecated(although there is no indication for that). This is an artefact from old lsp-mode. I will add deprecation notice.